classes/class-wc-connect-compatibility-wc30.php

```diff
@@ -130,6 +130,33 @@ public function get_product_name_from_order( $product_id, $order ) {
 			return sprintf( __( '#%d - [Deleted product]', 'woocommerce-services' ), $product_id );
 		}
 
+		/**
+		 * For a given product ID, it tries to find its price inside an order's line items.
+		 *
+		 * @param int $product_id Product ID or variation ID
+		 * @param WC_Order $order
+		 * @return float The product (or variation) price, or NULL if it wasn't found
+		 */
+		public function get_product_price_from_order( $product_id, $order ) {
+			foreach ( $order->get_items() as $line_item ) {
+				$line_product_id = $line_item->get_product_id();
+				$line_variation_id = $line_item->get_variation_id();
+
+				if ( ! $line_product_id ) {
+					$line_product_id = (int) get_metadata( 'order_item', $line_item->get_id(), '_product_id', true );
+				}
```

Comment:
Why are these "backup" retrievals necessary?

Reply:
I just copied the logic from `get_product_name_from_order()`, I assume it was like this for a good reason :/

Reply:
Alright - it would be great if we knew why it was done in the first place - does this jog your memory at all? https://github.com/Automattic/woocommerce-services/commit/ec4e72a72fde21789a5570bd23237d1ddcf09907

Reply:
Wow, it was me... plot twist!
Seriously, I don't have any recollection of that code. But I wouldn't have added that fallback if I hadn't encountered a case where it made sense. I can't find what it could possibly be now.